app/assets/javascripts/tsheets_integration.js

```diff
@@ -0,0 +1,65 @@
+var tsheets = (function($) {
```

Comment:
I should admit that I'm not very good at JS, i can write it. But i don't really know how to handle big JS codebase properly, it always seems like a mess. I don't currently see big reasons to switch to react, it offers a lot of functionality -- so I'm maybe missing something. I never did react projects, i just played around with it. I found a library that has all the best parts of react, but is much more sane and minimalistic - http://riotjs.com . In one project then we considered using react, we picked riotjs and never had a need to have something more then that.